nmt_test.go

```diff
@@ -773,3 +774,97 @@ func generateRandNamespacedRawData(total int, nidSize int, leafSize int) [][]byt
 func sortByteArrays(src [][]byte) {
 	sort.Slice(src, func(i, j int) bool { return bytes.Compare(src[i], src[j]) < 0 })
 }
+
+func TestMinNamespace(t *testing.T) {
+	type testCase struct {
+		name string
+		tree *NamespacedMerkleTree
+		want namespace.ID
+	}
+	testCases := []testCase{
+		{
+			name: "example tree with four leaves",
+			tree: exampleTreeWithFourLeaves(),
+			want: namespace.ID{0},
+		},
+		{
+			name: "example tree with eight leaves",
+			tree: exampleTreeWithEightLeaves(),
+			want: namespace.ID{1, 1},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.tree.MinNamespace()
+			assert.Equal(t, tc.want, got)
+		})
+	}
+}
+
+func TestMaxNamespace(t *testing.T) {
+	type testCase struct {
+		name string
+		tree *NamespacedMerkleTree
+		want namespace.ID
+	}
+	testCases := []testCase{
+		{
+			name: "example tree with four leaves",
+			tree: exampleTreeWithFourLeaves(),
+			want: namespace.ID{3},
+		},
+		{
+			name: "example tree with eight leaves",
+			tree: exampleTreeWithEightLeaves(),
+			want: namespace.ID{8, 8},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.tree.MaxNamespace()
```

Comment:
[optional] It's totally up to you, but I wanted to suggest that combining these two tests i.e., `TestMaxNamespace` and `TestMinNamespace` might be a good idea. The test prep for both of them is pretty much identical, which means we would only need to build one set of NMTs.  Also and the logic of the two methods are very similar. This could potentially give a slightly higher performance in our tests.

Reply:
Yea good point. I almost did this originally but I like when each method (e.g. `Method`) has it's own unit test (e.g. `TestMethod`) so that it's easy to navigate from method to unit test and see what the expected behavior is like. Since that isn't the case for other unit test names in this file, I'll consolidate.

Reply:
Resolved in https://github.com/celestiaorg/nmt/pull/105/commits/893ff0375a789dad850c9714b0b6a7993695e5ad